x-pack/plugins/apm/public/components/shared/charts/latency_chart/index.tsx

```diff
@@ -34,18 +37,30 @@ const options: Array<{ value: LatencyAggregationType; text: string }> = [
 
 export function LatencyChart({ height }: Props) {
   const history = useHistory();
+  const theme = useTheme();
+  const comparisonChartTheme = getComparisonChartTheme(theme);
   const { urlParams } = useUrlParams();
-  const { latencyAggregationType } = urlParams;
+  const { latencyAggregationType, comparisonEnabled } = urlParams;
   const license = useLicenseContext();
 
   const {
     latencyChartsData,
     latencyChartsStatus,
   } = useTransactionLatencyChartsFetcher();
 
-  const { latencyTimeseries, anomalyTimeseries, mlJobId } = latencyChartsData;
+  const {
+    currentPeriod,
+    previousPeriod,
+    anomalyTimeseries,
+    mlJobId,
+  } = latencyChartsData;
+
+  const timeseries = [
+    currentPeriod,
+    comparisonEnabled ? previousPeriod : undefined,
+  ].filter((data) => data) as Array<APMChartSpec<Coordinate>>;
```

Comment:
nit: instead of manually typing you could use a utility function that removes null/undefined and returns the right type

```suggestion
  function filterNil<T>(value?: T | null): value is T {
    return !isNil(value);
  }

  const timeseries = [
    currentPeriod,
    comparisonEnabled ? previousPeriod : undefined,
  ].filter(filterNil);
```

Reply:
Btw what is `previousPeriod`